public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -154,6 +155,15 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
     return queries.filter((query) => isExpressionQueryInAlert(query));
   }, [queries]);
 
+  useEffect(() => {
+    // we only remove or add the reducer(optimize reducer) expression when creating a new alert.
+    // When editing an alert, we assume the user wants to manually adjust expressions and queries for more control and customization.
+
+    if (!editingExistingRule && isOptimizeReducerEnabled) {
+      dispatch(optimizeReduceExpression({ updatedQueries: dataQueries, expressionQueries }));
+    }
+  }, [dataQueries, expressionQueries, editingExistingRule, isOptimizeReducerEnabled]);
+
```

Comment:
As this should work only for new rules, shouldn't we do that in `getDefaultQueries` instead? 
Doing it in `getDefaultQueries` could lower the complexity of this component

Reply:
we want to work also while editing the query, not only in the initial render

Reply:
That's true, but we already have code responsible for that, don't we?

Reply:
This `useEffect` dispatches `optimizeReduceExpression` on every keystroke. 
I guess we should do that only if the user switches between simple and advanced modes, wdyt?
I'm concerned that invoking that on every key stroke/input change will lead to errors and performance issues